config/routes.rb

```diff
@@ -1,9 +1,13 @@
 Rails.application.routes.draw do
-  get "/auth/:provider/callback", to: "sessions#omniauth"
-  post "/auth/:provider/callback", to: "sessions#omniauth"
-  get "/auth/:provider", to: redirect { |params, request| "/auth/#{params[:provider]}" }, as: :auth_provider
+  # get "/auth/:provider/callback", to: "sessions#omniauth"
+  # post "/auth/:provider/callback", to: "sessions#omniauth"
+  # get "/auth/:provider", to: redirect { |params, request| "/auth/#{params[:provider]}" }, as: :auth_provider
 
   scope "(:locale)", locale: /en|vi/ do
+    devise_for :users, controllers: {
```

Comment:
```suggestion
    devise_for :users, only: [:sessions, :registrations, :confirmations], controllers: {
```
Giới hạn các router theo tính năng